### PR TITLE
Fixes #437 simplify usage of dimension factory

### LINKED ISSUE #437: Simplify usage of DimensionFactory and RHS management

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
After Core update, removing  support for older version of MoBi project v 3.x

File: src/MoBi.Core/Domain/Model/MoBiContext.cs [hunk 1]
```diff
@@ -197,7 +196,6 @@ public T Resolve<T>()
 
       public void LoadFrom(IMoBiProject project)
       {
-         DimensionFactory.ProjectFactory = project.DimensionFactory;
```
Comment (msevestre) [inline src/MoBi.Core/Domain/Model/MoBiContext.cs]: No more of this non sense everywhere

File: src/MoBi.Core/Domain/UnitSystem/MoBiDimensionFactory.cs [hunk 2]
```diff
@@ -2,189 +2,55 @@
...
 using System.Linq;
 using MoBi.Assets;
-using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.UnitSystem;
 
 namespace MoBi.Core.Domain.UnitSystem
 {
    public interface IMoBiDimensionFactory : IDimensionFactory
    {
-      IDimensionFactory BaseFactory { get; }
```
Comment (msevestre) [inline src/MoBi.Core/Domain/UnitSystem/MoBiDimensionFactory.cs]: No more of this non sense everywhere

File: src/MoBi.Core/Domain/UnitSystem/MoBiDimensionFactory.cs [hunk 3]
```diff
@@ -2,189 +2,55 @@
...
-      ///    Returns the <see cref="IDimension" /> named <paramref name="dimensionName" /> or the default <c>NO_DIMENSION</c> if
-      ///    not found
-      /// </summary>
-      IDimension TryGetDimension(string dimensionName);
-
       IDimension TryGetDimensionCaseInsensitive(string dimensionName);
    }
 
-   public class MoBiDimensionFactory : IMoBiDimensionFactory
+   public class MoBiDimensionFactory : DimensionFactory, IMoBiDimensionFactory
```
Comment (msevestre) [inline src/MoBi.Core/Domain/UnitSystem/MoBiDimensionFactory.cs]: look at that... it inherits from the base class.. youhou!

File: src/MoBi.Core/Domain/UnitSystem/MoBiMergedDimensionFactory.cs [hunk 4]
```diff
@@ -1,48 +1,8 @@
...
-using System.Collections.Generic;
-using System.Linq;
-using MoBi.Assets;
-using OSPSuite.Core.Domain;
+﻿using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.UnitSystem;
 
 namespace MoBi.Core.Domain.UnitSystem
 {
-   public class MoBiMergedDimensionFactory : DimensionFactory
```
Comment (msevestre) [inline src/MoBi.Core/Domain/UnitSystem/MoBiMergedDimensionFactory.cs]: no more of this

File: src/MoBi.Core/Serialization/Services/PostSerializationStepsMaker.cs [hunk 5]
```diff
@@ -56,11 +49,6 @@ public void PerformPostDeserializationFor<T>(IReadOnlyList<T> deserializedObject
 
          //Convert loaded object to concentration if required
          deserializedObjects.Each(o => _amountToConcentrationConverter.Convert(o));
-
```
Comment (msevestre) [inline src/MoBi.Core/Serialization/Services/PostSerializationStepsMaker.cs]: gone

File: appveyor-coverage.yml [hunk 6]
```diff
@@ -4,6 +4,7 @@ image: Visual Studio 2019
 environment:
   MIK_TEX_INSTALL_DIR: 'C:\MikTex\'
   app_version: '9.0'
+  APPVEYOR_BLOCK_DOTNETCORE_TESTS_AUTORUN: true
```
Comment (Yuri05) [inline appveyor-coverage.yml]: what is this?

Comment (msevestre) [reply appveyor-coverage.yml]: Looking et the log in appveyor i saw the tthe tests were run twice. Once with nunit z and once with dotnet core.
This is a way to remove the behaviour

**Pull request merged**